lbry/lbry/extras/daemon/Components.py

```diff
@@ -136,86 +146,60 @@ def component(self):
                 log.warning("s3 appears to have corrupted header")
                 return
             final_size_after_download = response.content_length + local_header_size
-            write_mode = "wb"
             if local_header_size > 0:
                 log.info("Resuming download of %i bytes from s3", response.content_length)
-                write_mode = "a+b"
-            with open(self.headers_file, write_mode) as fd:
-                while True:
-                    chunk = await response.content.read(512)
-                    if not chunk:
-                        break
-                    fd.write(chunk)
-                    self._headers_progress_percent = math.ceil(
-                        float(fd.tell()) / float(final_size_after_download) * 100
-                    )
-            log.info("fetched headers from s3, now verifying integrity after download.")
-            self._check_header_file_integrity()
-
-    def local_header_file_height(self):
-        return max((self.local_header_file_size() / HEADER_SIZE) - 1, 0)
+            buffer, header_size = b'', self.headers.header_size
+            async for chunk in response.content.iter_any():
+                chunk = buffer + chunk
+                remaining = len(chunk) % header_size
+                chunk, buffer = chunk[:-remaining], bytes(chunk[-remaining:])
+                if not chunk:
+                    continue
+                if not await self.headers.connect(len(self.headers), chunk):
+                    log.warning("Error connecting downloaded headers from at %s.", self.headers.height)
+                    return
+                self._headers_progress_percent = math.ceil(
+                    float(self.headers.bytes_size) / float(final_size_after_download) * 100
+                )
```

Comment:
There are two places where the progress_percent is calcualted and they are slightly different. Can we have one place? Also, it should clip total to 0-100 range via `max(0, min(100, ...))`